src/main/java/use_cases/add_edit_expenses_use_case/ExpenseUCI.java

```diff
@@ -9,8 +9,16 @@
 public class ExpenseUCI implements ExpenseIB {
     private final ExpenseOB expenseOB;
```

Comment:
I noticed in both CategoryUCI and ExpenseUCI there are a lot of attributes saved. I think you should look at how many uses these have and get rid of the ones that are only used in 1 or 2 methods! I think the reason many of them are here is because you use them in private methods, but I recommend passing them in instead. Anyone who uses our code and tries to understand how the method works would likely get confused by all of these attributes.

Reply:
Make sure you keep the Factory and Presenter and anything used in both, but some attributes like name and value are likely not necessary to store here!